faker/tests/__init__.py

```diff
@@ -459,6 +459,16 @@ def test_date_time_this_period_with_tzinfo(self):
             datetime.datetime.now(utc).replace(second=0, microsecond=0)
         )
 
+    def test_binary(self):
+        from faker.providers.misc import Provider
+
+        for _ in range(999):
+            length = random.randint(0, 2 ** 10)
+            binary = Provider.binary(length)
+
+            self.assertTrue(isinstance(binary, bytes))
```

Comment:
I think this will fail on Python2, since os.urandom return `str`. You should compare with `six.binary_type` instead:

`self.assertTrue(isinstance(binary, six.binary_type))`